Automattic/jetpack
# Move RecordBarMeter to js-packages

#### Changes proposed in this Pull Request:
After the suggestions in #23522, this PR:

- Moves `RecordMeterBar` component from `plugins/jetpack/_inc/client/components` to `js-packages/components` as discussed in this p2 discussion - pcNPJE-zh-p2#comment-1113.
- Converts `RecordMeterBar` to TypeScript after moving it to js-packages.
- Creates a story for `RecordMeterBar` in TypeScript.
- Adds Unit tests for it written in TypeScript.
- Converts to TypeScript, the new `BarChart` component and related components that reside inside `plugins/jetpack/_inc/client/at-a-glance/backup-upgrade`.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

- Please follow the testing instructions for #23374 to ensure there is no regression.
- `cd projects/js-packages/storybook` and run `npm run storybook:dev` and confirm that you see `RecordMeterBar` component in Storybook.
- Run build, watch commands and unit tests to look for any regression
